feature/account/oauth/src/main/kotlin/app/k9mail/feature/account/oauth/domain/usecase/CheckIsGoogleSignIn.kt

```diff
@@ -0,0 +1,23 @@
+package app.k9mail.feature.account.oauth.domain.usecase
+
+import app.k9mail.feature.account.oauth.domain.DomainContract.UseCase
+
+internal class CheckIsGoogleSignIn : UseCase.CheckIsGoogleSignIn {
+    override fun execute(hostname: String): Boolean {
```

Comment:
This should only return `true` if hostname is one of the hostnames specified in `AppOAuthConfigurationFactory.createGmailConfiguration()`, i.e. `imap.gmail.com`, `imap.googlemail.com`, `smtp.gmail.com`, or `smtp.googlemail.com`.

Reply:
We're going to change the OAuthConfiguration to validate the hostnames from there.